includes/RemovePIIJob.php

```diff
@@ -0,0 +1,525 @@
+					],
+					'where' => [
+						'rev_edit_user_id' => $userId
+					]
+				]
+			],
+			'moderation' => [
+				[
+					'fields' => [
+						'mod_header_xff' => '',
+						'mod_header_ua' => '',
+						'mod_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'mod_user' => $userId
+					]
+				],
+				[
+					'fields' => [
+						'mod_header_xff' => '',
+						'mod_header_ua' => '',
+						'mod_ip' => '0.0.0.0',
+						'mod_user_text' => $userNewName
+					],
+					'where' => [
+						'mod_user_text' => $userOldName
+					]
+				]
+			],
+			'report_reports' => [
+				[
+					'fields' => [
+						'report_user_text' => $userNewName
+					],
+					'where' => [
+						'report_user_text' => $userOldName
+					]
+				],
+				[
+					'fields' => [
+						'report_handled_by_text' => $userNewName
+					],
+					'where' => [
+						'report_handled_by_text' => $userOldName
+					]
+				]
+			],
+			'Vote' => [
+				[
+					'fields' => [
+						'vote_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'vote_actor' => $userActorId
+					]
+				]
+			],
+			'wikiforum_category' => [
+				[
+					'fields' => [
+						'wfc_added_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wfc_added_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wfc_edited_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wfc_edited_actor' => $userActorId
+					]
+				],
+			],
+			'wikiforum_forums' => [
+				[
+					'fields' => [
+						'wff_last_post_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wff_last_post_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wff_added_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wff_added_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wff_edited_user_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'wff_edited_actor' => $userActorId
+					]
+				],
+			],
+			'wikiforum_replies' => [
+				[
+					'fields' => [
+						'wfr_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wfr_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wfr_edit_user_ip' => '0.0.0.0',
+					],
+					'where' => [
+						'wfr_edit_actor' => $userActorId
+					]
+				],
+			],
+			'wikiforum_threads' => [
+				[
+					'fields' => [
+						'wft_user_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'wft_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wft_edit_user_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'wft_edit_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wft_closed_user_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'wft_closed_actor' => $userActorId
+					]
+				],
+				[
+					'fields' => [
+						'wft_last_post_user_ip' => '0.0.0.0'
+					],
+					'where' => [
+						'wft_last_post_actor' => $userActorId
+					]
+				]
+			],
+		];
+
+		foreach ( $tableDeletions as $key => $value ) {
+			if ( $dbw->tableExists( $key ) ) {
```

Comment:
this (and all other database calls) should include `__METHOD__` for easier debugging. depending on the table sizes also consider chunking with `waitForReplicas()` calls in between

Reply:
Oh right thanks!

Reply:
Also do you mean `MediaWikiServices::getInstance()->getDBLoadBalancerFactory()->waitForReplication()`? Or is there actually a `waitForReplicas()` (if so I've just not heard of it but I might be wrong)

Reply:
Yeah, oops, it's `waitForReplication()`, but if you operate on some other database than the specific wikis main one you need to pass the database name to the function. I'm used to working with CA which aliases it as [waitForReplicas()](https://github.com/wikimedia/mediawiki-extensions-CentralAuth/blob/28e1ec49febb43d5ea79e9266bcd79f0ee150af1/includes/CentralAuthUtilityService.php#L92), sorry for the wrong name.